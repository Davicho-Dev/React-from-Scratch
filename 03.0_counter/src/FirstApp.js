import PropTypes from 'prop-types'

// export const FirstApp = ( { title, subtitle = `I'm subtitle` } ) => {
export const FirstApp = ( { title, subtitle } ) => {

	// if ( !greeting ) throw new Error( 'The greeting is necessary!' )

	return <>
		<h1>{ title }</h1>
		<hr/>
		<p>{ subtitle }</p>
	</>
}

FirstApp.propTypes = {
	subtitle: PropTypes.string, title: PropTypes.string.isRequired,

}

FirstApp.defaultProps = {
	subtitle: `I'm subtitle`,
}
