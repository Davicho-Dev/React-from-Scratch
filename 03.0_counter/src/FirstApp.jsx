import PropTypes from 'prop-types'

const FirstApp = ( { title, subTitle, name } ) =>
	(
		<>
			<h1 data-testid="test-title">{ title }</h1>
			<p>{ subTitle }</p>
			<p>{ subTitle }</p>
			<p>{ name }</p>
		</>
	)

FirstApp.propTypes = {
	title: PropTypes.string.isRequired,
	subTitle: PropTypes.string,
}

FirstApp.defaultProps = {
	name: 'Fernando Herrera',
	subTitle: 'No hay subtítulo',
	// title: 'No hay título',
}

export default FirstApp
