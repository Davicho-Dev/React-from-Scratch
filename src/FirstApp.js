import PropTypes from 'prop-types'

export const FirstApp = ( { title, subtitle } ) => {
	return <>
		<h1>{ title }</h1>
		<hr/>
		<p>{ subtitle }</p>
	</>
}

FirstApp.propTypes = {
	subtitle: PropTypes.string, title: PropTypes.string.isRequired,

}

FirstApp.defaultProps = {
	subtitle: `I'm subtitle`,
}
