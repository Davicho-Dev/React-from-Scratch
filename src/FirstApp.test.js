
import { shallow } from 'enzyme'
import React from 'react'

import { FirstApp } from './FirstApp'

describe( 'Test in FirstApp', () => {
	test( 'It must display <FirstApp />correctly.', () => {
		// const greeting = `Hi, i'm Goku`

		// const { getByText } = render( <FirstApp title={ greeting }/> )

		// wrapper.getByText()

		// expect( getByText( greeting + '!' ) ).toBeInTheDocument()
		const greeting = `Hi, i'm Goku`

		const wrapper = shallow(<FirstApp title={greeting}/>)

		expect( wrapper ).toMatchSnapshot()

	} )
} )
