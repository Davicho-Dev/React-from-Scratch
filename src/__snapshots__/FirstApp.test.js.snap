// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in FirstApp It must display <FirstApp />correctly. 1`] = `
<Fragment>
  <h1>
    Hi, i'm Goku
  </h1>
  <hr />
  <p>
    I'm subtitle
  </p>
</Fragment>
`;
